packages/terra-clinical-onset-picker/src/OnsetPicker.jsx

```diff
@@ -0,0 +1,389 @@
+    this.changeDate = this.changeDate.bind(this);
+    this.handleDateChange = this.handleDateChange.bind(this);
+
+    const ageValues = OnsetUtils.onsetToAge(this.props.birthdate, this.props.onsetDate);
+
+    this.state = {
+      granularity: this.props.granularity,
+      precision: this.props.precision,
+      onsetDate: this.props.onsetDate ? moment(this.props.onsetDate) : moment(),
+      age: ageValues.age,
+      ageUnit: ageValues.ageUnit,
+    };
+  }
+
+  /*
+   * Change state for granularity
+   */
+  changeGranularity(event) {
+    if (event.target.value === GranularityOptions.AGE) { // Calculate age values and update onsetDate to match age calculation
+      const ageValues = OnsetUtils.onsetToAge(this.props.birthdate, this.state.onsetDate);
+      const newDate = moment(this.props.birthdate).add(ageValues.age, ageValues.ageUnit);
+
+      this.setState({
+        age: ageValues.age,
+        ageUnit: ageValues.ageUnit,
+        onsetDate: newDate,
+      });
+
+      this.handleDateChange(newDate);
+    }
+    this.setState({ granularity: event.target.value });
+
+    if (this.props.granularitySelectOnChange) {
+      this.props.granularitySelectOnChange(event.target.value);
+    }
+  }
+
+  /*
+   * Change state for precision
+   */
+  changePrecision(event) {
+    this.setState({ precision: event.target.value });
+
+    if (this.props.precisionSelectOnChange) {
+      this.props.precisionSelectOnChange(event.target.value);
+    }
+  }
+
+  /*
+   * Change state for the age when using AGE granularity, and update onset date
+   */
+  changeAge(event) {
+    const newDate = moment(this.props.birthdate).add(Number(event.target.value), this.state.ageUnit);
+
+    this.setState({
+      age: Number(event.target.value),
+      onsetDate: newDate,
+    });
+    this.handleDateChange(newDate);
+  }
+
+  /*
+   * Change state for duration when using age granularity, and update onset date
+   */
+  changeAgeUnit(event) {
+    const newDate = moment(this.props.birthdate).add(this.state.age, event.target.value);
+
+    this.setState({
+      ageUnit: event.target.value,
+      onsetDate: newDate,
+    });
+    this.handleDateChange(newDate);
+  }
+
+  /*
+   * Update onset date when year changes
+   */
+  changeYear(event) {
+    let newDate = moment(this.state.onsetDate).year(event.target.value);
+    const newMonths = OnsetUtils.allowedMonths(this.context.intl, this.props.birthdate, newDate);
+
+    // Check if new onset month is available, otherwise change month to first possible month
+    if (newMonths.filter(month => parseInt(month.value, 10) === newDate.month()).length === 0) {
+      newDate = moment(newDate).month(newMonths[0].value);
+    }
+
+    this.setState({
+      onsetDate: newDate,
+    });
+    this.handleDateChange(newDate);
+  }
+
+  /*
+   * Update onset date when month changes
+   */
+  changeMonth(event) {
+    const newDate = moment(this.state.onsetDate).month(event.target.value);
```

Comment:
You would want to update this since this.state.onsetDate is now a moment object.

Reply:
No longer wrapping a moment object as a moment object.

Reply:
There are still a few places in render that needs to remove the wrapper as well as OnsetUtils.onsetToAge and OnsetUtils. allowedMonths.

Reply:
Thank you for double checking me Ben

Reply:
What I meant in my previous comment is that there are still 3 places in the render function that need to be updated:
https://github.com/cerner/terra-clinical/pull/184/files#diff-38eebe943861385618dfe5a92abc83bcR322
https://github.com/cerner/terra-clinical/pull/184/files#diff-38eebe943861385618dfe5a92abc83bcR333
https://github.com/cerner/terra-clinical/pull/184/files#diff-38eebe943861385618dfe5a92abc83bcR346

Reply:
Ok... now it should be taken care of.